### PR TITLE
Create dog mutation

### LINKED ISSUE #30: Add mutation to create a dog
Add a createDog mutation with the following arguments:
- name (required)
- activity level (required)
- weight (required)
- breed (required)
- birthdate (required)
- short description
- long description

### PR DESCRIPTION
**Pull request by chakeresa**
**Changes proposed in this pull request:**
 - Adds a `createDog` mutation and tests thereof
 - Adds a `DogInput` GraphQL type
 - Alters test helper methods to remove unused arguments and not compare ID for some dog tests

Resolves #30, related to user story #56 

**The following checks have been completed:**
 - [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
 - [x] Checked `coverage/index.html` - did not add any new code that's not covered by testing (if possible)
 - [x] Merged in the latest master to my branch with `git pull origin master` & resolved merge conflicts
 - [x] Ran `rails db:migrate`
 - [x] Ran the test suite - all tests are passing (or maybe skipped)
 - [x] Checked affected endpoints in Postman / GraphiQL
 - [x] Updated README for changes (new endpoints, new gems, etc)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,6 +79,16 @@ Object types are templates for resources in the database.  Each object type has
 * email - String (required)
 * googleToken - String (required)
 
+#### DogInputType Attributes
```
Comment (alexander-mathieu) [inline README.md]: 🐕 🐩 👍

File: app/graphql/mutations/create_dog.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module Mutations
+  class CreateDog < BaseMutation
+    null true
+
+    argument :dog, Types::DogInput, required: true
+
+    field :dog, Types::DogType, null: true
+
+    def resolve(dog:)
+      boot_unauthenticated_user
```
Comment (alexander-mathieu) [inline app/graphql/mutations/create_dog.rb]: 👢

File: app/graphql/types/mutation_type.rb [hunk 3]
```diff
@@ -2,5 +2,7 @@ module Types
   class MutationType < Types::BaseObject
     field :authenticate_user, mutation: Mutations::AuthenticateUser
     field :log_out_user, mutation: Mutations::LogOutUser
+
```
Comment (alexander-mathieu) [inline app/graphql/types/mutation_type.rb]: Is this space to prevent a conflict?

File: README.md [hunk 4]
```diff
@@ -249,6 +259,57 @@ Expected response:
 }
 ```
 
+#### createDog(dog: <DogInputType>)
+
+Creates a dog in the database for the current user (based on the `google_token` in the params). Returns a DogType object.
+
+Required arguments include:
+* name - String
```
Comment (alexander-mathieu) [inline README.md]: Should the required arguments be listed as only `dog:`?  These are the required arguments for the DogInputType.
